src/lib/helpers/ArrowHelper.svelte

```diff
@@ -0,0 +1,52 @@
+<script lang="ts">
+  import type { ArrowHelperProperties, Object3DProperties } from '$lib/types/components';
+  import { ArrowHelper as ThreeArrowHelper } from 'three'
+  import Object3DInstance from '../instances/Object3DInstance.svelte'
+
+  // Object3DInstance
+  export let position: Object3DProperties['position'] = undefined
+  export let scale: Object3DProperties['scale'] = undefined
+  export let rotation: Object3DProperties['rotation'] = undefined
+  export let lookAt: Object3DProperties['lookAt'] = undefined
+  export let viewportAware: Object3DProperties['viewportAware'] = false
+  export let inViewport: Object3DProperties['inViewport'] = false
+  export let castShadow: Object3DProperties['castShadow'] = undefined
+  export let receiveShadow: Object3DProperties['receiveShadow'] = undefined
+  export let frustumCulled: Object3DProperties['frustumCulled'] = undefined
+  export let renderOrder: Object3DProperties['renderOrder'] = undefined
+  export let visible: Object3DProperties['visible'] = undefined
+  
+  //self
+  export let dir:ArrowHelperProperties['dir'] = undefined;
+  export let origin:ArrowHelperProperties['origin'] = undefined;
+  export let length:ArrowHelperProperties['length'] = undefined;
+  export let color:ArrowHelperProperties['color'] = undefined;
+  export let headLength:ArrowHelperProperties['headLength'] = undefined;
+  export let headWidth:ArrowHelperProperties['headWidth'] = undefined;
+
+  let helper: ThreeArrowHelper = new ThreeArrowHelper (dir,origin,length,color,headLength,headWidth);
+
+  $:{
```

Comment:
The direction can change too, account for reactivity.
Currently the property `dir` is of type `THREE.Vector3`. Threlte uses an extension to that type called `Position`

```ts
export type Position =
  | Vector3
  | {
      x?: number
      y?: number
      z?: number
    }
```

so that a user can also just pass *some* values and others are assumed 0. I'd say that the property `dir` is a good fit for that type as well, but we need to account for that of course.
On top of that [the docs](https://threejs.org/docs/index.html?q=arrowhelper#api/en/helpers/ArrowHelper.setDirection) state that the vector passed into `setDirection()` needs to be a unit vector, so it needs to be normalized.

Reply:
Another option would be to lay out the component types like this:

```ts
export type ArrowHelperProperties = Omit<Object3DInstanceProperties, 'object'> & {
  target?: Position,
  color?: ColorRepresentation,
  headLength?: number,
  headWidth?: number
}
```
And calculate the property `length` based on the supplied `position` against `target`. Three.js is clear about how they calculate the default values for `headLength` and `headWidth` so I guess we can roll with that too. What do you think?
It's a bit further away from the original three.js implementation and that is something I'm very careful about. In this case it might make setting up the component easier depending on the use case.

Reply:
I believe Helpers should be able to be set up very straightforward so the implementation you're suggesting here is very appealing.
Honestly, I don't have the 3D/Math knowledge to infer the `dir` unit vector from an `origin` and a `target` Vector but I could implement the length.